src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -3736,21 +3729,26 @@ void  MacroAssembler::set_narrow_klass(Register dst, Klass* k) {
 
 // Maybe emit a call via a trampoline. If the code cache is small
 // trampolines won't be emitted.
-address MacroAssembler::trampoline_call(Address entry) {
+address MacroAssembler::patchable_far_call(Address entry) {
```

Comment:
It doesn't look nice to me for `UseTrampolines` checks to be spread across this function. I would suggest to keep the original `MacroAssembler::trampoline_call` and let `MacroAssembler::patchable_far_call` delegate work to it under `UseTrampolines`. What do you think?

Reply:
fixed